external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -289,6 +293,13 @@ private void doSeekRetriableTopicPartitions() {
                 kafkaConsumer.seek(rtp, acked.get(rtp).getCommittedOffset() + 1);    // Seek to last committed offset
             }
         }
+        
+        if(!retriableTopicPartitions.isEmpty()) {
+            //Pause other partitions for this poll so maxUncommittedPartitions isn't exceeded by too much
```

Comment:
I am still not quite understanding why we need to pause all other topic partitions in the case of a retrial. This could potentially lead to the scenario that one topic partition may be causing a lot of failures, but all others are running fine. If that's the case, we will the throttling all other topic-partitions for no particular reason.

Are you doing this pause to handle the case when `nunUncommittedOffsets` may exceed `maxUncommittedOffsets`? If that is the case, then in the case of a retrial we should only only when  `nunUncommittedOffsets >= maxUncommittedOffsets`. Otherwise perhaps we should poll from all topic-partitions.